libs/hooks/src/use-dev-mode.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useState } from 'react';
+
+interface TUseDevModeReturn {
+  devMode: boolean;
+  propagateDevMode: (_isDevMode: boolean) => void;
+}
+
+/**
+ * Hook to check or set Dev Mode preference
+ * @example useDevMode hook
+ * ```typescript
+ * const { devMode, propagateDevMode } = useDevMode();
+ * ```
+ * @returns `{ devMode, propagateDevMode }` The current devMode value and a function to set the devMode preference.
+ * The `propagateDevMode` function takes just a boolean parameter indicating whether or not you are enabling dev mode.
+ **/
+export const useDevMode = (): TUseDevModeReturn => {
+  // get the dev mode preference, if any, from local storage
+  const isDevMode = window.localStorage.getItem('DevMode');
+
+  const [devMode, setDevMode] = useState<boolean>(isDevMode === 'true');
+
+  const propagateDevMode = useCallback((_isDevMode: boolean) => {
```

Comment:
there is no need for a react hook, the page needs to full refresh after setting the dev mode